src/main/scala/org/hatdex/hat/api/json/DataStatsFormat.scala

```diff
@@ -16,57 +16,74 @@ import play.api.libs.json._
+  implicit protected val dataDebitFormat: Format[DataDebit]                                      = RichDataJsonFormats.dataDebitFormat
+  implicit protected val dataDebitEventFormat: Format[DataDebitEvent]         = Json.format[DataDebitEvent]
+  implicit protected val dataDebitOperationFormat: Format[DataDebitOperation] = Json.format[DataDebitOperation]
+
+  implicit val mapReads: Reads[scala.collection.immutable.HashMap[String, Long]] =
+    new Reads[scala.collection.immutable.HashMap[String, Long]] {
+      def reads(jv: JsValue): JsResult[scala.collection.immutable.HashMap[String, Long]] = {
+        val fields = jv.as[JsObject].fields
+
+        val s: collection.immutable.Seq[(String, Long)] = fields.map {
+          case (k, v) => k -> v.as[Long]
+        }.toSeq
+
+        // jv.as[JsObject].fields.map {
+        //   case (k, v) =>
+        //     k -> v.as[Long]
+        // }
+        JsSuccess(scala.collection.immutable.HashMap[String, Long](s: _*))
       }
-      JsSuccess(HashMap[String, Long](fields: _*))
     }
-  }
 
-  implicit val mapWrites: Writes[HashMap[String, Long]] = new Writes[HashMap[String, Long]] {
-    def writes(map: HashMap[String, Long]): JsValue =
-      Json.obj(map.map {
-        case (s, o) =>
-          val ret: (String, JsValueWrapper) = s.toString -> JsNumber(o)
-          ret
-      }.toSeq: _*)
-  }
+  implicit val mapWrites: Writes[scala.collection.immutable.HashMap[String, Long]] =
+    new Writes[scala.collection.immutable.HashMap[String, Long]] {
+      def writes(map: scala.collection.immutable.HashMap[String, Long]): JsValue =
+        Json.obj(map.map {
+          case (s, o) =>
+            val ret: (String, JsValueWrapper) = s.toString -> JsNumber(o)
+            ret
+        }.toSeq: _*)
+    }
 
   implicit val mapFormat: Format[HashMap[String, Long]] = Format(mapReads, mapWrites)
 
   implicit val endpointStatsFormat: Format[EndpointStats] = Json.format[EndpointStats]
 
-  implicit protected val dataCreditStatsFormat: Format[DataCreditStats]       = Json.format[DataCreditStats]
-  implicit protected val dataStorageStatsFormat: Format[DataStorageStats]     = Json.format[DataStorageStats]
-  implicit protected val inboundDataStatsFormat: Format[InboundDataStats]     = Json.format[InboundDataStats]
-  implicit protected val outboundDataStatsFormat: Format[OutboundDataStats]   = Json.format[OutboundDataStats]
-  implicit protected val richDataDebitFormat                                  = RichDataJsonFormats.richDataDebitFormat
-  implicit protected val dataDebitFormat                                      = RichDataJsonFormats.dataDebitFormat
-  implicit protected val dataDebitEventFormat: Format[DataDebitEvent]         = Json.format[DataDebitEvent]
-  implicit protected val dataDebitOperationFormat: Format[DataDebitOperation] = Json.format[DataDebitOperation]
+  // implicit protected val dataCreditStatsFormat: Format[DataCreditStats] = Json.format[DataCreditStats]
```

Comment:
Is there a reason we leave commented code in?

Reply:
No, I just moved that block due to a forward reference.